### PR TITLE
Fix Kahuna push notification swizzling

### PR DESCRIPTION
**Pull request by f2prateek**
cc @sharekris @TorreyKahuna

### CODE REVIEW

File: Analytics/Integrations/Kahuna/SEGKahunaIntegration.h [hunk 1]
```diff
@@ -22,6 +22,7 @@
 @property (nonatomic) NSDictionary *pushInfo;
 @property (nonatomic) UIApplicationState applicationState;
 @property (nonatomic) BOOL kahunaInitialized;
```
Comment (f2prateek) [inline Analytics/Integrations/Kahuna/SEGKahunaIntegration.h]: @sharekris I noticed that you made these properties atomic in your change, is that strictly required?

Comment (sharekris) [inline Analytics/Integrations/Kahuna/SEGKahunaIntegration.h]: The property is being accessed in Start (which is in background thread) as well as App Delegate notifications (which is in Main thread). So made the properties atomic.

File: Analytics/Integrations/Kahuna/SEGKahunaIntegration.m [hunk 2]
```diff
@@ -396,6 +396,7 @@ - (void)swizzleAppDelegateMethods
 - (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
 {
     @try {
+        [[SEGKahunaPushMonitor sharedInstance] failedToRegisterPush:error];
         if ([SEGKahunaPushMonitor sharedInstance].kahunaInitialized) {
             [KahunaAnalytics handleNotificationRegistrationFailure:error];
         }
```
Comment (sharekris) [inline Analytics/Integrations/Kahuna/SEGKahunaIntegration.m]: This "if ([SEGKahunaPushMonitor sharedInstance].kahunaInitialized) {" should not be present. The "failedToRegisterPush" takes care of the IF condition internally.

Comment (f2prateek) [inline Analytics/Integrations/Kahuna/SEGKahunaIntegration.m]: updated

File: Analytics/Integrations/Kahuna/SEGKahunaIntegration.h (hunk 1 above)
Comment (f2prateek) [inline Analytics/Integrations/Kahuna/SEGKahunaIntegration.h]: Thanks, updated the properties to be atomic.

**Pull request merged**